Statistics/Histograms/Histogram.cs

```diff
@@ -481,6 +567,34 @@ public bool TestForConvergence(double upperq, double lowerq)
             }
             return _Converged;
         }
+        public bool Equals(IHistogram histogramToCompare)
```

Comment:
This is probably fine. You could also just override the Equals operator. I think it would be public override bool Equals... and see what intellisence gives you.